src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -525,14 +556,20 @@ ActStatus ActionHandlers::sec_mode_cmd_to_ue(SM::ControlBlock& cb)
 	nas.header.security_encryption_algo = ue_ctxt->getUeSecInfo().getSelectSecAlg();
 	nas.header.security_integrity_algo = ue_ctxt->getUeSecInfo().getSelectIntAlg();
 	nas.header.nas_security_param = AUTHREQ_NAS_SECURITY_PARAM;
-	const uint8_t num_nas_elements = SEC_MODE_NO_OF_NAS_IES;
+        uint8_t num_nas_elements = SEC_MODE_NO_OF_NAS_IES;
+	/* UE additional security capability is replayed in sec_mode_command when MME supports the handling of the same
+         * Spec 24.301 v 15.6.0  sec:8.2.20.6*/
+	if(mme_cfg->feature_list.dcnr_support && ue_ctxt->getUeAddSecCapabPres())
+    	{
+            num_nas_elements += 1;
+    	}
 	nas.elements = (nas_pdu_elements *) calloc(num_nas_elements, sizeof(nas_pdu_elements)); // TODO : should i use new ?
```

Comment:
@asreek  should we avoid using calloc now in Mme cpp files ? Can you please guide us ?

Reply:
Hi Ajay,
We can explore using mem pool managed MsgBuffer objects for encoding q message structures. Will discuss on this during next call.
Thanks,
Anjana